core/src/main/java/org/axonframework/queryhandling/QueryGateway.java

```diff
@@ -132,6 +133,93 @@
+                                 query,
+                                 ResponseTypes.instanceOf(initialResponseType),
+                                 ResponseTypes.instanceOf(updateResponseType),
+                                 SubscriptionQueryBackpressure.defaultBackpressure());
+    }
+
+    /**
+     * Sends given {@code query} over the {@link QueryBus} and returns result containing initial response and
+     * incremental updates (received at the moment the query is sent, until it is cancelled by the caller or closed by
+     * the emitting side).
+     *
+     * @param query               The {@code query} to be sent
+     * @param initialResponseType The initial response type used for this query
+     * @param updateResponseType  The update response type used for this query
+     * @param <Q>                 The type of the query
+     * @param <I>                 The type of the initial response
+     * @param <U>                 The type of the incremental update
+     * @return registration which can be used to cancel receiving updates
+     */
+    default <Q, I, U> SubscriptionQueryResult<I, U> subscriptionQuery(Q query, ResponseType<I> initialResponseType,
+                                                                      ResponseType<U> updateResponseType) {
+        return subscriptionQuery(query.getClass().getName(),
+                                 query,
+                                 initialResponseType,
+                                 updateResponseType,
+                                 SubscriptionQueryBackpressure.defaultBackpressure());
+    }
+
+    /**
+     * Sends given {@code query} over the {@link QueryBus} and returns result containing initial response and
+     * incremental updates (received at the moment the query is sent, until it is cancelled by the caller or closed by
+     * the emitting side).
+     *
+     * @param queryName           A {@link String} describing query to be executed
+     * @param query               The {@code query} to be sent
+     * @param initialResponseType The initial response type used for this query
+     * @param updateResponseType  The update response type used for this query
+     * @param backpressure        The backpressure mechanism to deal with producing of incremental updates
+     * @param <Q>                 The type of the query
+     * @param <I>                 The type of the initial response
+     * @param <U>                 The type of the incremental update
+     * @return registration which can be used to cancel receiving updates
+     */
+    <Q, I, U> SubscriptionQueryResult<I, U> subscriptionQuery(String queryName, Q query,
+                                                              ResponseType<I> initialResponseType,
+                                                              ResponseType<U> updateResponseType,
+                                                              SubscriptionQueryBackpressure backpressure);
```

Comment:
Just wondering, why is there an explicit back-pressure parameter? Isn't back pressure already available in the SubscriptionQueryResult?

Reply:
When creating a `Flux` for updates, we can provide some other back pressure mechanism instead of `FluxSink.OverflowStrategy.BUFFER` which is the default.

Regarding the back pressure on `SubscriptionQueryResult`, I guess that you are talking about fine-grained back pressure by requesting certain amount of updates from the emitting side (emitting side can choose not to respect that)? 

The first one is client side back pressuring, and the other one is communicating the feedback about consuming pace on client side.

Hope this explanation answers your concern :)